clip/tokenizer.py

```diff
@@ -0,0 +1,128 @@
+# Copyright © 2023 Apple Inc.
+
+import json
+from pathlib import Path
+from typing import Any, Union
+
+import mlx.core as mx
+import regex
+from huggingface_hub import hf_hub_download
+
+
+class CLIPTokenizer:
```

Comment:
Why not just use the `ClipTokenizer` from Transformers directly instead of copying the code here?

Reply:
This is a good question. You could also ask 'Why not just use the `CLIPImageProcessor` from `transformers` instead of implementing the new one? My goal was to remove dependency on `transformers` by implementing simple and hackable implementations of the tokenizer and the image processor. These implementations are 'good enough' for demo purposes. In particular, their outputs are equal to the corresponding `transformers` outputs up to numerical accuracy.  Please see the tests.

We can remove custom implementations, but we will need `transformers`. I do not have a strong opinion regarding this. Please suggest whether we want simpler, custom implementations or a dependency on `transformers`.